tox.ini

```diff
@@ -95,13 +95,20 @@ deps =
     devinfra: git+https://github.com/matplotlib/pytest-mpl.git
     devinfra: git+https://github.com/astropy/pytest-astropy.git
 
-# The following indicates which [project.optional-dependencies] from pyproject.toml will be installed
+    # Duplicates test_all in pyproject.toml due to upstream bug
+    # https://github.com/tox-dev/tox/issues/3433
+    # But we cannot use alldeps because it messed up oldest-deps pins.
+    devdeps-!noscipy: objgraph>=1.6.0
+    devdeps-!noscipy: skyfield>=1.20
+    devdeps-!noscipy: sgp4>=2.3
+    devdeps-!noscipy: array-api-strict>=1.0
+
+# The following indicates which [project.optional-dependencies] from pyproject.toml will be installed.
+# test_all does not work here due to upstream bug https://github.com/tox-dev/tox/issues/3433
 extras =
     test: test
-    recdeps: recommended
-    recdeps: test_all
+    recdeps: all
```

Comment:
I don't know what recdeps is for. It tries to install test_all anyway, so might as well be alldeps because test_all already includes recommended. But maybe somewhere downstream is using it, so I'll keep it for now.

Reply:
Wait, are you saying that our tox envs are considered public API somehow ? This feels daunting.

Reply:
Not public per se, but it is not in scope in this PR to break a possible usage somewhere.

Reply:
I'm puzzled. How would it be used anywhere else ?

Reply:
Looks like @mhvk added it back in https://github.com/astropy/astropy/pull/10464  and I really don't remember why I added `test_all` to it in https://github.com/astropy/astropy/pull/13651

Reply:
We don't use it in docs either. So I think it is okay to remove but let's do that in a different PR.

Reply:
Actually I change my mind. It is cleaner to remove it here, rather than changing the specs. Thanks for bringing this up!

Reply:
My logic was that since we have a set of recommended dependencies, we probably wanted some tests to just run with those. And at least at the time, I thought that sometimes I just want the "standard" stuff like `scipy` to be tested, but not have to wait to pull everything else in. But in practice, I certainly hadn't remembered it and so haven't used it - probably enough reason to remove it...  If kept, though, just use `recdeps: recommended`.

Reply:
I should have made a note on why I added test_all to it because I have zero recollection. I am neutral on keeping it but reverting back to recommended, or just removing it. I don't use tox locally.